transform/flatten_mangler.go

```diff
@@ -271,3 +288,41 @@ func getUnderlyingKindType(t reflect.Type) (reflect.Kind, reflect.Type) {
 	}
 	return k, t
 }
+
+func stripPtrs(val reflect.Value) reflect.Value {
+	for val.IsValid() {
+		switch val.Kind() {
+		case reflect.Ptr, reflect.Interface:
+			val = val.Elem()
+		default:
+			return val
+		}
+	}
+	return val
+}
+
+// GetField takes in a reflect.Value and returns the value of the field at the
+// given field path or the zero value if it's not populated
+func GetField(v reflect.Value, t reflect.Type, fieldPath string) interface{} {
```

Comment:
Let's return a `reflect.Value` so we don't need to convert back if we're doing anything interesting with the value.

Reply:
Good call, will make it much easier to use in the flag package. Fixed